nixos/modules/system/boot/modprobe.nix

```diff
@@ -58,14 +58,9 @@ with lib;
 
     environment.systemPackages = [ pkgs.kmod ];
 
-    system.activationScripts.modprobe = stringAfter ["specialfs"]
-      ''
-        # Allow the kernel to find our wrapped modprobe (which searches
-        # in the right location in the Nix store for kernel modules).
-        # We need this when the kernel (or some module) auto-loads a
-        # module.
-        echo ${pkgs.kmod}/bin/modprobe > /proc/sys/kernel/modprobe
-      '';
+    systemd.services.systemd-modules-load.preStart = ''
+      echo ${pkgs.kmod}/bin/modprobe > /proc/sys/kernel/modprobe
```

Comment:
Is this also too late? I seem to recall the kernel being able to load modules itself via the current value of `/proc/sys/kernel/modprobe`, which can happen arbitrarily early I think?

Reply:
If it can really happen arbitrarily early, then this should probably already be done in the initrd. I had a case where a test failed, but setting this as preStart for systemd-modules-load fixed it.

Reply:
This should probably happen before both systemd-modules-load and systemd-udev. These are the only two things that I can think of that would load kernel modules before sysinit.target